src/сomponents/Footer.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  todos: Todo[];
+  filter: Filter;
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>;
+  handleClearCompleted: () => void;
+  loading: boolean;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filter,
+  setFilter,
+  handleClearCompleted,
+  loading,
+}) => {
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+
+  const filters = Object.values(Filter).map(value => ({
+    value,
+    label: value,
+    href: `#/${value.toLowerCase()}`,
+    dataCy: `FilterLink${value}`,
+  }));
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${activeTodos} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        {filters.map(({ value, label, href, dataCy }) => (
+          <a
+            key={value}
+            href={href}
+            className={`filter__link ${filter === value ? 'selected' : ''}`}
```

Comment:
Don't forget to use classnames library everywhere for add classes with condition